appgate/openapi/openapi.py

```diff
@@ -118,7 +124,61 @@ def entity_names(
 
 
 def generate_crd(entity: Type, short_names: Dict[str, str]) -> str:
+    prev_default_object_fields = settings.default_object_fields
+
+    def attrs_fields(cls: type) -> Optional[Sequence[ObjectField]]:
+        if hasattr(cls, "__attrs_attrs__"):
+            obj: List[ObjectField] = []
+            for a in getattr(cls, "__attrs_attrs__"):
```

Comment:
You can use the `attrs.fields` function for this, see https://www.attrs.org/en/stable/api.html#attrs.fields